src/core/interpreter.cc

```diff
@@ -84,13 +84,20 @@ void RedisTranslator::OnString(std::string_view str) {
   ArrayPost();
 }
 
-// Doubles are not supported by Redis, however we can support them.
-// Here is the use-case:
-// local foo = redis.call('zscore', 'myzset', 'one')
-// assert(type(foo) == "number")
 void RedisTranslator::OnDouble(double d) {
+  static constexpr double kConvertEps = std::numeric_limits<double>::epsilon();
+
+  double fractpart, intpart;
+  fractpart = modf(d, &intpart);
```

Comment:
I'm not sure epsilon is correct if the fractional part is taken from a large number. Does it hold up to max long?

Reply:
They scale in the example for comparing the difference,  https://en.cppreference.com/w/cpp/types/numeric_limits/epsilon and in my mind the modf cant improve on accuracy if a large number is already stored skewed

But I checked in the range of int64-max and it seems that modf(max_int) < double_eps 🔦

Reply:
lgtm